src/mango/test/mango.py

```diff
@@ -120,6 +120,8 @@ def recreate(self):
                     # db exists and it is empty -- exit condition is met
                     return
                 self.delete()
+            else:
+                r.raise_for_status()
```

Comment:
If we're going to raise, we'll exit the try loop immediately?

Reply:
Yes, that is right.  I think that is a meaningful action here since it probably makes no sense to keep trying if a valid HTTP error is raised.

Reply:
It would depend on what return codes we expect to get here. The retry loop was added to handle [intermittent failures](https://github.com/apache/couchdb/commit/0414ef3e5d39de159cb6e985110773ea5522aea4) most so we wouldn't want to bring back flakiness here.

Reply:
That is a good idea to make a distinction between recoverable and non-recoverable errors here.  In my case, it was HTTP 401 which is non-recoverable since this is a user configuration error (bad credentials).  Similar ones could be 402, 403, 412, 413 for example.

Reply:
Yeah, that's basically what might cause issues. For instance we start the test with a deleted db so we get a 404 and would immediately stop the test, while the intent was for us to retry and recreate the db and continue. There may be other non-fatal status codes to retry as well (slow auth propagation? others?) while some maybe fatal and shouldn't be retried.

Maybe a simpler solution may be to shorten the retry timeout or the total number of retries so on failure it just doesn't take that long?

Reply:
The retry logic here already implements an exponential back-off.  I see no reason to trim it down.

Reply:
Oh, sorry — that is just a linear back-off.  But what I wanted to say here is that we are already increasing the timeout on each try.

Reply:
Normally I don't remember seeing an issue with mango tests behaving this way in CI. I think it must because you're trying to run them separately on a different cluster? Maybe what we can have a pre-test check for auth and ability to talk to the cluster in general, and if not quickly exit before the test even starts?